spec/models/poll_spec.rb

```diff
@@ -2,5 +2,63 @@
 
 RSpec.describe Poll, type: :model do
     it { should validate_presence_of(:title) }
-    it { should have_many(:candidates)}
+    it { should have_many(:candidates).dependent(:destroy) }
+    it { should accept_nested_attributes_for(:candidates) }
+
+    describe "#winning_candidates" do
+        let(:poll) { Poll.create!(title: "Poll", start_time: "2022-09-29T16:31", end_time: "2022-09-29T14:34") }
+        let(:candidate_one) { Candidate.create!(title: "Candidate one", poll: poll) }
+        let(:candidate_two) { Candidate.create!(title: "Candidate two", poll: poll) }
+        let(:candidate_two) { Candidate.create!(title: "Candidate three", poll: poll) }
+
+        subject { poll }
+
+        context "when there is a winner" do
+            let!(:vote1) { Vote.create!(candidate: candidate_one) }
+            let!(:vote2) { Vote.create!(candidate: candidate_one) }
+            let!(:vote3) { Vote.create!(candidate: candidate_one) }
+            let!(:vote4) { Vote.create!(candidate: candidate_two) }
+
+            before do
+                poll.candidates.each do |candidate|
+                    candidate.calculate_result
+                end
+            end
+
+            it "calculates the winning candidate" do
+                expect(subject.winning_candidates).to include(candidate_one)
```

Comment:
Probably worth checking how many `winning_candidates` you got back here too, this will increase your confidence.